test/formal/combine_byte_shl.py

```diff
@@ -0,0 +1,27 @@
+from rule import Rule
+from opcodes import *
+
+"""
+byte(A, shl(B, X))
+given A < 32 && B == 256 - 8 * (A + 1)
+->
+and(X, 0xff)
+"""
+
+rule = Rule()
+
+n_bits = 256
+
+# Input vars
+X = BitVec('X', n_bits)
+A = BitVec('A', n_bits)
+B = BitVec('B', n_bits)
+
+# Non optimized result
+nonopt = BYTE(A, SHL(B, X))
+# Optimized result
+opt = AND(X, 0xff)
+
+rule.require(ULT(A, 32) and B == 256 - 8 * (A + 1))
```

Comment:
```suggestion
rule.require(And(ULT(A, 32), B == 256 - 8 * (A + 1)))
```
This should be a z3 Bool and not a Python Bool

Reply:
(At least that's what my z3 is complaining about)

Reply:
After this /\ the proof goes through locally for me